src/clock.ml

```diff
@@ -89,8 +89,49 @@ let leave (s : active_source) =
     List.iter (log#important "%s")
       (Pcre.split ~pat:"\n" (Printexc.get_backtrace ()))
 
-(** Base clock class. *)
-class clock id =
+(** {1 Clock implementation}
+  * This was formerly known as the Root.
+  * One could think of several clocks for isolated parts of a script.
+  * One can also think of alsa-clocks, etc. *)
+
+let conf = Dtools.Conf.void ~p:(Configure.conf#plug "root") "Streaming clock settings"
+
+let conf_max_latency =
+  Dtools.Conf.float ~p:(conf#plug "max_latency") ~d:60. "Maximum latency in seconds"
+    ~comments:
+      [ "If the latency gets higher than this value, the outputs will be reset,";
+        "instead of trying to catch it up second by second.";
+        "The reset is typically only useful to reconnect icecast mounts." ]
+
+(** Timing stuff, make sure the frame rate is correct. *)
+
+let time = Unix.gettimeofday
+
+let usleep d =
+  (* In some implementations,
+   * Thread.delay uses Unix.select which can raise EINTR.
+   * A really good implementation would keep track of the elapsed time and then
+   * trigger another Thread.delay for the remaining time.
+   * This cheap thing does the job for now.. *)
+  try Thread.delay d with Unix.Unix_error (Unix.EINTR, _, _) -> ()
+
+(** In [`CPU] mode, synchronization is governed by the CPU clock.
```

Comment:
I not sure about this comment: above, you say that cpu corresponds to "cpu burning mode" where you produce as fast as possible, but this comment sounds like you use the cpu clock to achieve something like realtime.

Reply:
I'm trying to clarify the terminology. "cpu burning mode" is now gone, `CPU` means just that the CPU clock (I believe the one on the motherboard) is driving synchronization by keeping track of the time elapsed while producing each frame and sleeping or accelerating accordingly.